src/classes/groundplaces.ts

```diff
@@ -1,66 +1,110 @@
+import { Storage } from '../classes/storage';
+import { WebServices } from './webservices';
 import {
   AutoComplete,
-  StopGroup,
-  StopCluster,
+  SegmentProviderStop,
   StopGroupGpuid,
   StopClusterGpuid,
-  StopGroupProperties,
-  StopClusterProperties,
+  CreateStopGroupProperties,
+  CreateStopClusterProperties,
+  UpdateStopProperties,
   AutoCompleteFilters,
   GroundPlacesDiff,
   GroundPlaceType,
+  GroundPlaces,
 } from '../types';
 
 /**
  * @description GroundPlaces business logic.
  */
-export class GroundPlaces {
+export class GroundPlacesController {
+  private readonly storageService: Storage;
+
+  private readonly webService: WebServices;
+
+  /**
+   * @description Manipulate GroundPlaces file.
+   * @param {string} groundPlacesFile - The file to manipulate, can only be JSON for now.
+   */
+  constructor() {
+    this.storageService = new Storage();
+
+    this.webService = new WebServices();
+  }
+
+  /**
+   * @description Init GroundPlaces file.
+   * @param {string|undefined} groundPlacesFile - The file to manipulate, can only be JSON for now.
+   * If you provide empty value, the default file will be the file retrieved from Amazon S3.
+   */
+  public init(groundPlacesFile?: string): void {
+    if (groundPlacesFile) {
+      this.storageService.initFile(groundPlacesFile);
+    } else {
+      const groundPlacesFileS3 = this.webService.downloadDistantGroundPlacesMaster();
+
+      this.storageService.initFile(groundPlacesFileS3);
+    }
+  }
+
   /**
    * @description Returns a list of places.
    * @param {string} query - Can be a name, a Gpuid, a unique name or other name.
    * @param {AutoCompleteFilters} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
-   * @returns {AutoComplete}
+   * @returns {AutoComplete[]}
    */
   // @ts-ignore
-  public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete {}
+  public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete[] {}
 
   /**
-   * @description Create the stopGroup with the values given.
+   * @description Create a new StopGroup from a SegmentProviderStop.
+   * @param {SegmentProviderStop} segmentProviderStop - The SegmentProviderStop on which is based the StopGroup.
+   * @param {CreateStopGroupProperties} stopGroupProperties - Properties that are needed to create a new StopGroup.
    * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier of the stopCluster parent.
-   * @param {StopGroup} stopGroupInfos - StopGroup informations.
-   * @returns {StopGroup}
+   * @returns {void}
    */
-  // @ts-ignore
-  public createStopGroup(stopClusterGpuid: StopClusterGpuid, stopGroupInfos: StopGroup): void {}
+  public createStopGroup(
+    segmentProviderStop: SegmentProviderStop,
+    stopGroupProperties: CreateStopGroupProperties,
```

Comment:
not great the naming. 
Why not smth like : 
`createStopGroupParams`
with a segmentProviderStop in it, the stopclusterGpuid and the properties independnat? 

@rimiti @Theo-Le-Donne

Reply:
not sure what you mean. Put the three params in a interface called CreateStopGroupParams ?